presto-main/src/main/java/io/prestosql/sql/planner/IdentitySensitiveExpressionNodeInliner.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
> This behavior is caused by `SubqueryPlanner` who collects correlated symbols
using equality comparison, and as a result equates different occurrences
of the symbols.

`NodeRef` was introduced later than `SubqueryPlanner` IIRC.